ct-app/core/components/peer_session_management.py

```
@@ -0,0 +1,47 @@
+from core.api.response_objects import Session
+
+
+class PeerSessionManagement:
+
+    def __init__(self, session: Session, ip: str):
+        self.session = session
+        self.ip = ip
+        try:
+            self.socket = self.create_socket()
+        except (socket.error, ValueError) as e:
+            raise ValueError(f"Error while creating socket: {e}")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)